toolkit/tools/graphanalytics/graphanalytics.go

```diff
@@ -0,0 +1,295 @@
+				return
+			}
+
+			dependencyName := nodeDependencyName(dependency)
+			insertIfMissing(srpmsBlockedBy, pkgSRPM, dependencyName)
+
+			return
+		})
+
+	}
+
+	printTitle("[INDIRECT] SRPMs closest to being ready to build")
+	printReversedMap(srpmsBlockedBy, "total unmet dependencies", maxResults)
+}
+
+// nodeDependencyName returns a common dependency name for a node that will be shared across similair Meta/Run/Build nodes for the same package.
+func nodeDependencyName(node *pkggraph.PkgNode) (name string) {
+	// Prefer the SRPM name if possible, otherwise use the unversioned package name
+	srpmName := filepath.Base(node.SrpmPath)
+	if srpmName == "" || srpmName == "<NO_SRPM_PATH>" {
+		name = node.VersionedPkg.Name
+	} else {
+		name = srpmName
+	}
+
+	return
+}
+
+// printTitle prints a formatted title
+func printTitle(title string) {
+	logger.Log.Info("")
+	logger.Log.Info("================================================")
+	logger.Log.Info(title)
+	logger.Log.Info("================================================")
+}
+
+// sortMap returns a sorted slice of a map. If inverse is set, it will return the smallest entries first.
+// It will sort entries by the number of values each key has. For keys with the same number of entries, it will
+// sort alphabetically.
+func sortMap(mapToSort map[string][]string, inverse bool) (pairList []mapPair) {
+	pairList = make([]mapPair, 0, len(mapToSort))
+	for key, value := range mapToSort {
+		pairList = append(pairList, mapPair{key, value})
+	}
+
+	sort.Slice(pairList, func(i, j int) bool {
+		if len(pairList[i].value) == len(pairList[j].value) {
+			return pairList[i].key < pairList[j].key
+		}
+
+		if inverse {
+			return len(pairList[i].value) < len(pairList[j].value)
+		}
+
+		return len(pairList[i].value) > len(pairList[j].value)
+	})
+
+	return
+}
+
+// insertIfMissing appens a value to the key in a map if it is not present.
+// Will alter data.
+func insertIfMissing(data map[string][]string, key string, value string) {
+	if sliceutils.Find(data[key], value) == -1 {
+		data[key] = append(data[key], value)
+	}
+}
+
+// printMap will sort and print the smallest entries, using the valueDescription in the format.
+func printReversedMap(data map[string][]string, valueDescription string, maxResults int) {
+	const inverse = true
+	pairList := sortMap(data, inverse)
+	printSlice(pairList, valueDescription, maxResults)
+}
+
+// printMap will sort and print the largest entries, using the valueDescription in the format.
+func printMap(data map[string][]string, valueDescription string, maxResults int) {
+	const inverse = false
+	pairList := sortMap(data, false)
+	printSlice(pairList, valueDescription, maxResults)
+}
+
+// printSlice prints the first maxResults entries, using the valueDescription in the format.
+func printSlice(pairList []mapPair, valueDescription string, maxResults int) {
+	if len(pairList) == 0 {
+		return
+	}
+
+	for i, pair := range pairList {
+		if i >= maxResults {
```

Comment:
>maxResults [](start = 10, length = 10)

I think this should be closer to `if maxResults != 0 && i >= ...`. According to the help text, zero should print us everything but here it prints nothing.